Add HOWTO on manually upgrading Shoot clusters (#182)
citynetwork/docs: docs/howto/kubernetes/gardener/rolling-upgrades.md

---
description: How to manually upgrade machine images or Kubernetes versions for Gardener clusters
---
# Conducting rolling upgrades

By default, Kubernetes clusters created with {{k8s_management_service}}
in {{brand}} are upgraded automatically. Those upgrades take place
during a specified maintenance window, and you may find out more about
[what they involve and how they
work](../../../../background/kubernetes/gardener/autoupgrades). Besides
the automatic upgrades, you can manually apply any upgrades available

> for your cluster. In the following two examples, we first show how to

Author: Ah, yes!

manually upgrade the machine image the cluster nodes are based on and
then how to upgrade the Kubernetes version the cluster uses.

## Upgrading machine images

In the left-hand vertical pane of the {{gui}}, select Containers →
[{{k8s_management_service}}](https://{{gui_domain}}/containers/gardener).
In the central pane, you will see all your {{k8s_management_service}}

> clusters, each on its own row. If a Kubernetes upgrade is available, you
> will notice, in the *Version* column, a triangular marker together with
> the new Kubernetes version. On the other hand, if there is a new machine
> image version for the cluster nodes, in the *Workers* column you will
> notice just a triangular marker --- no version number.

Reviewer: Somehow, this word order would sit better with me:

```suggestion
clusters, each on its own row. If a Kubernetes upgrade is available, you
will notice a triangular marker together with
the new Kubernetes version, in the *Version* column.
On the other hand, if there is a new machine
image version for the cluster nodes, you will
notice just a triangular marker --- no version number --- in the *Workers* column.
```

Author: Indeed, I think your version is *less* Christos-English. I am taking note of your suggestion and propose an equally acceptable version (I think).

![Garden Linux and Kubernetes upgrades available](assets/rollupgr-01.png)

To get more information about the new machine image available, click on
the cluster row to bring its details into view, then select the *Worker
Groups* tab. There, you will see the version of the new machine image.

> To start the upgrade process, click any of the two double-arrow-circle
> icons.

Reviewer: I *think* the more common name for this icon is "two circular arrows". But I think you can also just use the icon itself, like so:

```suggestion
To start the upgrade process, click any of the two :fontawesome-solid-arrows-rotate:
icons.
```

Author: Fontawesome --- nice!

To start the upgrade process, click any of the two

> ":fontawesome-solid-arrows-rotate:" icons.

Author: I generally try to avoid quotation marks. In this particular case, I used them because the fontawesome symbol looked somewhat weird --- at least to me. In any case, I am removing the marks.

The upgrade process will start immediately and will take some time to
complete. After it is done, there will be no triangular marker in the
*Workers* column, and the new image version will be displayed in the
*Worker groups* tab. Our example still has a triangular marker in the
*Version* column. That is because we have not upgraded Kubernetes yet.

![Garden Linux upgraded](assets/rollupgr-03.png)

## Upgrading Kubernetes

> To get the Kubernetes upgrade process going, click the orange three-dot
> round icon on the right of the cluster row. In the pop-up menu that

Reviewer: Maybe use:

```suggestion
To get the Kubernetes upgrade process going, click the orange :material-dots-horizontal-circle:
icon on the right of the cluster row. In the pop-up menu that
```

Author: Uhm, somehow I managed to miss that one :/

To get the Kubernetes upgrade process going, click the orange three-dot
round icon on the right of the cluster row. In the pop-up menu that
appears, you will notice an option named *Kubernetes Patch Available*.
That one is not always there, but now it is and acts as another
indicator of an available Kubernetes upgrade. Please select the option
by clicking on it.

![Kubernetes patch available](assets/rollupgr-04.png)

> A new pop-up window named *Upgrade Kubernetes Version for: cluster_name*

Reviewer: Please make it abundantly clear that `cluster_name` is to be read as a placeholder, like so:

```suggestion
A new pop-up window named *Upgrade Kubernetes Version for: <cluster_name>*
```

> A new pop-up window named *Upgrade Kubernetes Version for: cluster_name*

Reviewer: OK, then maybe this?

```suggestion
A new pop-up window named *Upgrade Kubernetes Version for: &lt;cluster_name&gt;*
```

Author: Aha, the HTML-entities-thing works!

> A new pop-up window named *Upgrade Kubernetes Version for: igloolik*
> appears, where *igloolik* is the name of our example cluster and a
> [really cool place](https://en.wikipedia.org/wiki/Igloolik). From the

Reviewer: I think this is charming, but I would still suggest:

```suggestion
A new pop-up window named *Upgrade Kubernetes Version for: <name>*
appears. From the
```

Author: OK --- and I really hope no Iglulingmiut will be reading this any time soon :/
